src/main/drivers/pwm_output.c

```diff
@@ -359,6 +369,22 @@ void pwmCompleteMotorUpdate(void)
 
 #ifdef USE_DSHOT
     if (isMotorProtocolDshot()) {
+
+        //Check if there are commands to send
+        if(!circularBufferIsEmpty(&commandsCircularBuffer) &&
+            micros() - lastCommandSent > DSHOT_COMMAND_INTERVAL_US){
```

Comment:
I'm not sure if this is the logic we want. It's safe to assume that we WANT all DSHOT commands to arrive to the ESC before continuing sending motor values. Also for robustness we should send each DSHOT command several times - 8? 16? This is something we need to define (or borrow from BF).

If we have commands pending - we should be sending only commands, no scheduling or interleaving.

Reply:
I think we should still keep the queue (in case we would want to send several different commands), but to save queue depth, it might be better to implement something like a static struct:

```
typedef struct {
    dshotCommands_e cmd;
    int remainingRepeats;
} currentExecutingCommand_t;
```

When we reach `remainingRepeats == 0` we look at the queue to check if there is a new command pending. If there is a command - we pop it, update the `cmd` and initialize `remainingRepeats` to repeat count.

WDYT?